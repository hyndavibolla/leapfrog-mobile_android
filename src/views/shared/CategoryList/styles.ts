import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE, MEASURE } from '../../../constants';

export const styles = {
  container: css`
    padding-vertical: ${MEASURE.MODAL_PADDING}px;
    padding-horizontal: 16px;
    flex: 1;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
  `,
  categoryContainer: css`
    flex-direction: column;
    justify-content: center;
  `,
  categoryItem: css`
    margin-top: 4px;
  `,
  headerContainer: css`
    padding-top: 8px;
  `,
  title: css`
    text-align: center;
    color: ${COLOR.BLACK};
    font-size: ${FONT_SIZE.SMALLER};
    margin-bottom: 24px;
  `,
  clearBtn: css`
    font-size: ${FONT_SIZE.SMALL};
    color: ${COLOR.PRIMARY_BLUE};
    margin-bottom: 16px;
  `,
  headerTitle: css`
    font-size: 14px;
    font-weight: bold;
    margin-right: 4px;
  `,
  footerContainer: css`
    padding: 20px 32px 0;
    align-items: center;
  `,
  innerButton: css`
    padding: 16px 0;
    height: 50px;
    width: 295px;
  `,
  buttonText: css`
    font-family: ${FONT_FAMILY.HEAVY};
  `
};
