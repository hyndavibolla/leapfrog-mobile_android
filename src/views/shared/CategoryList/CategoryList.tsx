import React, { memo } from 'react';
import { View, ScrollView, TouchableHighlight, ViewStyle } from 'react-native';

import { CategoryCard } from '_components/CategoryCard';
import { Text } from '_components/Text';
import { Button } from '_components/Button';

import { useTestingHelper } from '_utils/useTestingHelper';

import { ICategory } from '_models/mission';

import { FONT_FAMILY } from '_constants/styles';

import { styles } from './styles';

export interface Props {
  categoryList: ICategory[];
  selectedNameList: string[];
  onToggle: (categoryName: string) => void;
  onClearAll: () => void;
  onApply: () => void;
  style?: ViewStyle;
}

function CategoryList({ categoryList, onToggle, selectedNameList, onClearAll, onApply, style }: Props) {
  const { getTestIdProps } = useTestingHelper('category-list');

  return (
    <View style={[styles.container, style]}>
      <View style={styles.headerContainer}>
        <Text font={FONT_FAMILY.BOLD} style={styles.title}>
          FILTERS
        </Text>
        <TouchableHighlight underlayColor="transparent" onPress={onClearAll} {...getTestIdProps('clear-all')}>
          <Text font={FONT_FAMILY.BOLD} style={styles.clearBtn}>
            Clear all
          </Text>
        </TouchableHighlight>
      </View>
      <ScrollView>
        <View style={styles.categoryContainer}>
          {categoryList.map(({ name, imageUrl }) => (
            <View style={styles.categoryItem} key={name}>
              <CategoryCard image={imageUrl} label={name} isSelected={selectedNameList.includes(name)} onPress={() => onToggle(name)} />
            </View>
          ))}
        </View>
      </ScrollView>
      <View style={styles.footerContainer}>
        <Button innerContainerStyle={styles.innerButton} textStyle={styles.buttonText} onPress={onApply} {...getTestIdProps('apply')}>
          Apply
        </Button>
      </View>
    </View>
  );
}

export default memo(CategoryList);
