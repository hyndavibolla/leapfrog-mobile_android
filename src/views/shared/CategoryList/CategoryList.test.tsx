import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import CategoryList, { Props } from './CategoryList';

import { getMissionCategory_1, getMissionCategory_2 } from '_test_utils/entities';

describe('CategoryList', () => {
  let props: Props;

  const mockOnToggle = jest.fn();
  const mockOnClearAll = jest.fn();
  const mockOnApply = jest.fn();

  beforeEach(() => {
    props = {
      categoryList: [getMissionCategory_1(), getMissionCategory_2()],
      selectedNameList: [],
      onToggle: mockOnToggle,
      onClearAll: mockOnClearAll,
      onApply: mockOnApply
    };
  });

  it('should render ', () => {
    const { toJSON } = render(<CategoryList {...props} selectedNameList={[getMissionCategory_1().name]} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should not render any category card when the categoryList is empty', () => {
    props.categoryList = [];
    const { queryByTestId } = render(<CategoryList {...props} selectedNameList={[getMissionCategory_1().name]} />);
    expect(queryByTestId('category-card-container')).toBeNull();
  });

  it('should call onToggle callback when a card was pressed', () => {
    const { getAllByTestId } = render(<CategoryList {...props} />);
    fireEvent.press(getAllByTestId('category-card-container')[0]);
    expect(mockOnToggle).toBeCalledWith(getMissionCategory_1().name);
  });

  it('should call onClearAll callback when clear all was pressed', () => {
    const { getByTestId } = render(<CategoryList {...props} />);
    fireEvent.press(getByTestId('category-list-clear-all'));
    expect(mockOnClearAll).toBeCalled();
  });

  it('should call onApply callback when the apply button was pressed', () => {
    const { getByTestId } = render(<CategoryList {...props} />);
    fireEvent.press(getByTestId('category-list-apply'));
    expect(mockOnApply).toBeCalled();
  });
});
