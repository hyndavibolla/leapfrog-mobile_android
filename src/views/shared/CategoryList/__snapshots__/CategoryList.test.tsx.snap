// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList should render  1`] = `
<View
  style={
    Array [
      Object {
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "paddingTop": 8,
      }
    }
  >
    <Text
      font="SFProDisplay-Bold"
      style={
        Array [
          Object {
            "color": "#909BA0",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 16,
            "lineHeight": 22,
          },
          Object {
            "color": "#262C2F",
            "fontSize": 16,
            "marginBottom": 24,
            "textAlign": "center",
          },
        ]
      }
    >
      FILTERS
    </Text>
    <View
      accessibilityLabel="category-list-clear-all"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="category-list-clear-all"
    >
      <Text
        font="SFProDisplay-Bold"
        style={
          Array [
            Object {
              "color": "#909BA0",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 16,
              "lineHeight": 22,
            },
            Object {
              "color": "#0066CC",
              "fontSize": 18,
              "marginBottom": 16,
            },
          ]
        }
      >
        Clear all
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 4,
            }
          }
        >
          <View
            accessibilityLabel="category-card-container"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#EBEDF2",
                "borderBottomColor": "#0066CC",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderBottomWidth": 2,
                "borderLeftColor": "#0066CC",
                "borderLeftWidth": 2,
                "borderRightColor": "#0066CC",
                "borderRightWidth": 2,
                "borderStyle": "solid",
                "borderTopColor": "#0066CC",
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "borderTopWidth": 2,
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 20,
                "paddingTop": 4,
                "width": "100%",
              }
            }
            testID="category-card-container"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "flex-start",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "height": 50,
                          "marginRight": 8,
                          "resizeMode": "contain",
                          "width": 50,
                        },
                        false,
                      ],
                    ]
                  }
                >
                  <FastImageView
                    accessibilityLabel="image-with-fallback-container"
                    accessible={true}
                    onFastImageError={[Function]}
                    onFastImageLoadEnd={[Function]}
                    onFastImageLoadStart={[Function]}
                    resizeMode="cover"
                    source={
                      Object {
                        "uri": "https://cdn2.iconfinder.com/data/icons/bright-webshop/512/T-shirt-512.png",
                      }
                    }
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                    testID="image-with-fallback-container"
                  />
                </View>
                <Text
                  accessibilityLabel="category-card-text"
                  accessible={true}
                  ellipsizeMode="tail"
                  font="SFProDisplay-Bold"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#909BA0",
                        "fontFamily": "SFProDisplay-Bold",
                        "fontSize": 16,
                        "lineHeight": 22,
                      },
                      Array [
                        Object {
                          "color": "#262C2F",
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "fontSize": 18,
                        },
                        Object {
                          "color": "#0066CC",
                        },
                      ],
                    ]
                  }
                  testID="category-card-text"
                >
                  Apparel & Accessories
                </Text>
              </View>
              <Text
                accessibilityLabel="icon-text"
                accessible={true}
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#0066CC",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontFamily": "sywmax",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="icon-text"
              >
                î¥Š
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 4,
            }
          }
        >
          <View
            accessibilityLabel="category-card-container"
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#EBEDF2",
                "borderBottomColor": "#EBEDF2",
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderBottomWidth": 2,
                "borderLeftColor": "#EBEDF2",
                "borderLeftWidth": 2,
                "borderRightColor": "#EBEDF2",
                "borderRightWidth": 2,
                "borderStyle": "solid",
                "borderTopColor": "#EBEDF2",
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
                "borderTopWidth": 2,
                "flexDirection": "row",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 20,
                "paddingTop": 4,
                "width": "100%",
              }
            }
            testID="category-card-container"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexBasis": 0,
                    "flexDirection": "row",
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "justifyContent": "flex-start",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "height": 50,
                          "marginRight": 8,
                          "resizeMode": "contain",
                          "width": 50,
                        },
                        false,
                      ],
                    ]
                  }
                >
                  <FastImageView
                    accessibilityLabel="image-with-fallback-container"
                    accessible={true}
                    onFastImageError={[Function]}
                    onFastImageLoadEnd={[Function]}
                    onFastImageLoadStart={[Function]}
                    resizeMode="cover"
                    source={
                      Object {
                        "uri": "https://images.vexels.com/media/users/3/142247/isolated/lists/3f0b86857be6d4f07e2c257a045d576e-olla-de-perfume-plana.png",
                      }
                    }
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                    testID="image-with-fallback-container"
                  />
                </View>
                <Text
                  accessibilityLabel="category-card-text"
                  accessible={true}
                  ellipsizeMode="tail"
                  font="SFProDisplay-Bold"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#909BA0",
                        "fontFamily": "SFProDisplay-Bold",
                        "fontSize": 16,
                        "lineHeight": 22,
                      },
                      Array [
                        Object {
                          "color": "#262C2F",
                          "flexBasis": 0,
                          "flexGrow": 1,
                          "flexShrink": 1,
                          "fontSize": 18,
                        },
                        false,
                      ],
                    ]
                  }
                  testID="category-card-text"
                >
                  Beauty & Wellness
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingBottom": 0,
        "paddingLeft": 32,
        "paddingRight": 32,
        "paddingTop": 20,
      }
    }
  >
    <View
      accessibilityLabel="category-list-apply"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="category-list-apply"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#0066CC",
              "borderBottomLeftRadius": 30,
              "borderBottomRightRadius": 30,
              "borderTopLeftRadius": 30,
              "borderTopRightRadius": 30,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 8,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 8,
            },
            Object {
              "height": 50,
              "paddingBottom": 16,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 16,
              "width": 295,
            },
          ]
        }
      >
        <View
          style={false}
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#909BA0",
                  "fontFamily": "SFProDisplay-Medium",
                  "fontSize": 16,
                  "lineHeight": 22,
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 14,
                    "textAlign": "center",
                  },
                  Object {
                    "fontFamily": "SFProDisplay-Heavy",
                  },
                  undefined,
                ],
              ]
            }
          >
            Apply
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
